### DISCUSSION TITLE
Incorrect branch names in deployment history

### DISCUSSION THREAD

**Discussion by reidab** [community/community#39377] [Actions]
### Select Topic Area

Bug

### Body

I'm using Actions with environments set to auto-create deployments. This is populating the deployment history for our repo.

We're using a git-flow process, where our `main` branch represents our shipped code and our `develop` branch is the default branch used for development.

When deploying, we merge code from `develop` to `main`, then trigger a deployment from `main`.

The GitHub UI seems to be looking at the last commit on `main`, seeing that it's a PR from `develop`, and showing `develop` as the source branch in the deployment history, even though the commit it references (the merge of the `develop` -> `main` PR) isn't present on `develop`.

The underlying deployment record via the API correctly attributes the source of the deployment to `main`, but the UI shows `develop` instead.

<img width="666" alt="image" src="https://user-images.githubusercontent.com/13192/202591811-c4f9dfe8-713e-420c-996d-011073a679c4.png">

Is there any way to avoid this?

**Comment by joshft91** [comment]
I have the same issue. `main` kicked off the deploy workflow, but the name (`develop` in your screenshot) is set to `my-feature-branch-example` instead of showing `main` (in my opinion). I don't understand the reasoning to show an old branch name or PR# when the deployment was caused by a commit into `main`.

Here is another thread discussing this as well, I believe: https://github.com/orgs/community/discussions/17292
